test-packages/type-tests/test/v4/filters.spec.ts

```diff
@@ -10,3 +11,12 @@ and(
     any(TestEntityMultiLink.STRING_PROPERTY.equals('test'))
   )
 );
+
+// $ExpectType Filter<TestEntity, string>
+TestEntity.ENUM_PROPERTY.equals('Member1');
```

Comment:
Could you also add a test for integer based enum fixing what we expect. I know there were some limitation but a test is always good to fix our expected behaviour.

Reply:
or is this the test from above?

Reply:
Discussed, the number based enum is about the Option A and Option B in the ADR, which is not implemented.
There are tests to make sure numbers are not allowed to be used as parameter.